src/material-experimental/mdc-chips/chip-edit-input.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  Directive,
+  ElementRef,
+} from '@angular/core';
+
+/**
+ * A directive that makes a span editable and exposes functions to modify and retrieve the
+ * element's contents.
+ */
+@Directive({
+  selector: 'span[matChipEditInput]',
+  host: {
+    'class': 'mdc-chip__primary-action mat-chip-edit-input',
+    'role': 'textbox',
+    'tabindex': '-1',
+    'contenteditable': 'true',
+  },
+})
+export class MatChipEditInput {
+  constructor(
+      private readonly _elementRef: ElementRef,
```

Comment:
nit: this whole thing can be 1 line

Reply:
Done, though it's now three lines because we're injecting DOCUMENT.